### ISSUE TITLE
Can't get skore UI (launch) to work: `detail	"Not Found"`

### ISSUE THREAD

**Issue by GaelVaroquaux** [probabl-ai/skore#1003]
### Describe the bug

This is going to be a very non helpful bug report, and I hesitated submitting it, but I just can't get skore launch to do something useful.

After creating the project, I do "skore launch tmp", and skore does start, opening up a web browser, but I don't get the skore UI, I get the following instead:
![image](https://github.com/user-attachments/assets/830ab93f-78be-4e32-a6ec-f6a1f15e01d5)

### Steps/Code to Reproduce

Simplest way to reproduce (shell commands)
```
$ python3 -c "import skore; proj = skore.create('tmp'); proj.put('test', 1)"
$ skore launch tmp
```

### Expected Behavior

Getting the skore UI, or a useful error message if I did something wrong

### Actual Behavior

The skore server returns an error json, as shown above, with content:

```
{"detail":"Not Found"}
```
headers are:

```
content-length | 22
-- | --
content-type | application/json
date | Sun, 22 Dec 2024 15:28:54 GMT
server | uvicorn

Accept	text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding	gzip, deflate, br, zstd
Accept-Language	en-US,en;q=0.5
Connection	keep-alive
Host	localhost:22140
Priority	u=0, i
Sec-Fetch-Dest	document
Sec-Fetch-Mode	navigate
Sec-Fetch-Site	cross-site
Upgrade-Insecure-Requests	1
User-Agent	Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0
```

### Environment

```shell
System:
    python: 3.12.3 (main, Nov  6 2024, 18:32:19) [GCC 13.2.0]
executable: /usr/bin/python3
   machine: Linux-6.8.0-49-generic-x86_64-with-glibc2.39

Python dependencies:
        skore: 0.0.0+unknown
          pip: 24.0
   setuptools: 68.1.2
    diskcache: 5.6.3
      fastapi: 0.115.6
        numpy: 1.26.4
 plotly<6,>=5: None
      pyarrow: 16.1.0
         rich: 13.7.1
scikit-learn<1.7: None
        skops: 0.11.0
      uvicorn: 0.34.0
```
This is under Ubuntu 24.04.1 LTS.

I realize that the above info is actually not quite right (skore.show_version is failing), I have the following:

>>> sklearn.__version__
'1.6.0'
>>> plotly.__version__
'5.24.1'
```
```

**Comment by MarieS-WiMLDS** [comment]
hello Gaël, thanks for this issue and the others!  
This happens to me too when I forget to do make install the front-end (done thanks to `make build-skore-ui`). can you try to do that and launch the UI again please?  

@rouk1, is it possible to have another error message to help the user, even if the make build skore ui was not done?

About the problem for the versions, it's reported here: https://github.com/probabl-ai/skore/issues/987.

**Comment by GaelVaroquaux** [comment]
OK, this was indeed a lack of `make build-skore-ui`. Running this fixed the problem.

We need to improve installation users experience on the development version.
- Ideally, to have people from the Python community adopt things easily, we would not rely on a makefile, but on the Python build mechanism, which is what a vast majority of packages run
- An interim solution would have to have a "make install" that does all what is needed

**Comment by rouk1** [comment]
I can build a quick error page for devs. I agree that the makefile is not ideal as you'll have to install node anyway (I suppose that not all pythonistas did this...).

For future discussion what do you folks thinks of using some tools to have reproducible dev env ? (nix, pixi, ?)

**Comment by MarieS-WiMLDS** [comment]
Yes please Matthieu for the error page for devs after the milestone Skore 0.6!  
Do you need a design by Anass, or do you plan to do something very simple? (I'm not shocked by a json displaying only "please run make install-skore-ui").
